src/app/clusters/network-commissioning/network-commissioning.h

```diff
@@ -18,23 +18,84 @@
 
 namespace chip {
 namespace app {
 namespace clusters {
 namespace NetworkCommissioning {
-EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan operationalDataset,
-                                                                           uint64_t breadcrumb, uint32_t timeoutMs);
-EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan ssid,
-                                                                         ByteSpan credentials, uint64_t breadcrumb,
-                                                                         uint32_t timeoutMs);
-EmberAfNetworkCommissioningError OnEnableNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan networkID,
-                                                                        uint64_t breadcrumb, uint32_t timeoutMs);
+
+constexpr uint8_t kMaxNetworkIDLen       = 32;
+constexpr uint8_t kMaxThreadDatasetLen   = 254; // As defined in Thread spec.
+constexpr uint8_t kMaxWiFiSSIDLen        = 32;
+constexpr uint8_t kMaxWiFiCredentialsLen = 64;
+constexpr uint8_t kMaxNetworks           = 4;
+
+namespace Internal {
+enum class NetworkType : uint8_t
+{
+    kUndefined = 0,
+    kWiFi      = 1,
+    kThread    = 2,
+    kEthernet  = 3,
+};
+
+struct WiFiNetworkInfo
+{
+    uint8_t mSSID[kMaxWiFiSSIDLen + 1];
+    uint8_t mSSIDLen;
+    uint8_t mCredentials[kMaxWiFiCredentialsLen];
+    uint8_t mCredentialsLen;
+};
+
+struct NetworkInfo
+{
+    uint8_t mNetworkID[kMaxNetworkIDLen];
+    uint8_t mNetworkIDLen;
+    uint8_t mEnabled;
+    NetworkType mNetworkType;
+    union NetworkData
+    {
+        Thread::OperationalDataset mThread;
+        WiFiNetworkInfo mWiFi;
+    } mData;
+};
+
+enum class ClusterState : uint8_t
+{
+    kIdle            = 0,
+    kEnablingNetwork = 1,
+};
+
+constexpr uint8_t kInvalidNetworkSeq = 0xFF;
+
+void EncodeAndSendEnableNetworkResponse(CHIP_ERROR err);
+void SetEnablingNetworkSeq(uint8_t seq);
+ClusterState GetClusterState();
+CHIP_ERROR __attribute__((warn_unused_result)) MoveClusterState(ClusterState newState);
+void OnNetworkEnableStatusChangeCallback(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg);
+void SetCommissionerSecureSessionHandle(const SecureSessionHandle & handle);
+
+NetworkInfo * GetNetworkBySeq(uint8_t seq);
+} // namespace Internal
+
+void OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan operationalDataset, uint64_t breadcrumb,
+                                               uint32_t timeoutMs);
+void OnAddWiFiNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan ssid, ByteSpan credentials, uint64_t breadcrumb,
+                                             uint32_t timeoutMs);
+void OnEnableNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan networkID, uint64_t breadcrumb,
+                                            uint32_t timeoutMs);
```

Comment:
A bunch of this stuff needs to be documented.  For example, what does `SetEnablingNetworkSeq` do and what does the arg to it mean?

Also, for a bunch of these things, why are theyeven in the header, instead of just defined inside the .cpp file in an anonymous namespace?  They're only used inside the one .cpp file...

Reply:
These are caused by some code refactors (which are removed now) eariler. Moved these declares to the cpp file.